### ISSUE TITLE
Establish walkout date for leaving Github

### ISSUE THREAD

**Issue by Marak** [selfagency/microsoft-drop-ice#237]
Hello everyone.

I think it's great you've established this ultimatum, but I haven't found any indication for how long you've given Microsoft to make a decision.

It's important you establish a date in which you will all be leaving Github and transferring your projects. Without an organized deadline your threat of leaving just doesn't seem that serious.

**Comment by Marak** [comment]
@selfagency 

Please correct me if I am wrong, but to my understanding Satya Nadella has already responded to the petition by Microsoft staffers which you are referring to. Satya is categorically denying Microsoft from any wrongdoing.

Is there another petition you are referring to?

I've copy and pasted his response below for context.  

> Team,

> Like many of you, I am appalled at the abhorrent policy of separating immigrant children from their families at the southern border of the U.S. As both a parent and an immigrant, this issue touches me personally.

> I consider myself a product of two amazing and uniquely American things — American technology reaching me where I was growing up that allowed me to dream the dream and an enlightened immigration policy that then allowed me to live that dream. My story would not have been possible anywhere else.

> This new policy implemented on the border is simply cruel and abusive, and we are standing for change. Today Brad detailed our company's position on this issue, as well as the immigration legislation currently being considered in Congress, and I encourage you to read his blog post.

> I want to be clear: Microsoft is not working with the U.S. government on any projects related to separating children from their families at the border. Our current cloud engagement with U.S. Immigration and Customs Enforcement (ICE) is supporting legacy mail, calendar, messaging and document management workloads.

> Microsoft has a long history of taking a principled approach to how we live up to our mission of empowering every person and every organization on the planet to achieve more with technology platforms and tools, while also standing up for our enduring values and ethics. Any engagement with any government has been and will be guided by our ethics and principles. We will continue to have this dialogue both within our company and with our stakeholders outside.

> The immigration policy of this country is one of our greatest competitive advantages, and this is something we must preserve and promote. America is a nation of immigrants, and we're able to attract people from around the world to contribute to our economy, our communities and our companies. We are also a beacon of hope for those who need it the most. This is what makes America stronger. We will always stand for immigration policies that preserve every person's dignity and human rights. That means standing with every immigrant who works at Microsoft and standing for change in the inhumane treatment of children at the U.S. border today.

> Satya

This quote was taken from a Business Insider article [found here](http://www.businessinsider.com/microsoft-ceo-satya-nadella-memo-trump-family-separation-zero-tolerance-2018-6)

**Comment by Marak** [comment]
FYI, I think July 4th would be a good date to set the deadline considering it's America's Independence Day. 🇺🇸

**Comment by selfagency** [comment]
After Satya's letter, hundreds more Microsoft employees signed the petition. I don't think it's over yet.

**Comment by jerriep** [comment]
@selfagency Your pinned tweet states the following:

> My mother's father was in Auschwitz....

On the list of customers for GitLab is Bayer:

![image](https://user-images.githubusercontent.com/1006420/41825703-5352f7c6-784d-11e8-8518-af48b6de86f9.png)

Do you realise that Bayer was involved in the development of the gas used by the Nazis in the gas chambers? Do you realise that this company used Jewish slave labourers?

And let's not forget that Bayer now owns Monsanto. This is the company that developed Agent Orange. Ever seen what that stuff did to children?

May I suggest you make it very clear to your signees that they should not move to GitLab either - lest they are seen as hypocrites.

**Comment by selfagency** [comment]
The above is not only factually incorrect (Bayer's former parent company, IG Farben, had a minor stake in the company that made Zyklon B) but it is a continuation of your concern trolling that I already told you to take somewhere else. The fact that Bayer's parent company did something 70 years ago for which they apologized, paid DM 27,000,000 in restitution, were dissolved, and over which we have no control at this point in time, has absolutely nothing to do with the fact that there are human rights abuses happening **RIGHT NOW** that we can play a role in helping bring to an end. The same goes for Monsanto production of Agent Orange in the 1970s (also more than 40 years ago), for which they already settled a class action law suit paying restitution to the victims. That is not something we can undo. This is something we can stop, **NOW**. Finally, there is no ethical consumption under capitalism. There is nothing in this world that you can buy or use that isn't tainted by something evil in one way or another. Bearing that in mind, you have to pick your battles. This one seems to be pressing. Except to you, cuz you're a dick.

**Comment by selfagency** [comment]
Ah, it's all so clear now. Since Jerrie decided to go digging through my life, I decided to Google him back, and lo-and-behold, Jerrie is Microsoft fanboy who publishes a newsletter for .NET developers. I wonder why he'd be so upset with people trying to put pressure on his favorite company to stop supporting human rights abuses.

**Comment by Marak** [comment]
Can we please just delete the off-topic comments and stick to the original issue? I'm not sure all this name calling is helping with what you are trying to achieve here @selfagency 

Some of us have thousands of followers on Github and maintain projects which are downloaded millions of times per month. I think it would greatly help your cause if everyone here could all act with a bit more decorum.

**Comment by selfagency** [comment]
Frankly, I try not to be civil towards people whose only interest is to derail efforts to support the human rights of marginalized groups but I agree I'd rather not do it here.

**Comment by Marak** [comment]
@selfagency - 

I can think of a few ways to achieve this either using the Github API or by doing by it manually.

The easiest course of action might be opening a new issue and then using the `@username` feature of Github to send alerts to each user who has signed. Probably should only take 15 minutes. I don't think edits will work and too many alerts in a single message might cause a problem. Might want to make a new comment on the issue per every 25 users.

**Comment by selfagency** [comment]
that's a good idea.

**Comment by therobinkim** [comment]
The addition of a deadline changes the nature of the document, so I think you'll need to create an explicit opt-in strategy. A couple of ideas, for brainstorming purposes:

* Create a new section of the doc that says "these are the people that have agreed to leave by ____" and people can PR their names into that section.
* Create a separate branch or new repo where the deadline exists and people can PR their names into it.

**Comment by Marak** [comment]
Comrades, is there any status on this? 

I'm starting to doubt the resolve of the people who are supporting this action against Microsoft.

**It's time that the people on the list start showing Microsoft this is serious by migrating their projects and deleting their Github accounts.**

I'm certain if users who signed like like @sindresorhus, @jamiebuilds, and @seldo are the first move than others surely will follow by example.

**Comment by rbstrachan** [comment]
@Marak TL;DR - We should set a leaving date before asking prominent contributors and devs to permanently delete their accounts.

Let's not jump the gun. I am very serious, as I'm sure most of us are, about leaving GitHub. I have already migrated my projects over to GitLab in the event that Microsoft does not drop the contract, however, asking prominent contributors and developers like @sindresorhus @jamiebuilds and @seldo to permanently delete their accounts before a resolution has been met is slightly presumptuous.

I agree that a leaving date should be set but as is stands one has not. We should focus on that first before making irreversible decisions.

**Comment by edbedbe** [comment]
@rbstrachan The walkout date would definitely need to be communicated to the people who have signed, but I don't think that we should wait for the internal petition in Microsoft to show results. I'm doubtful that we'll see a public resolution anytime soon. Microsoft has only shown signs that they intend to downplay the ICE contract. 

I think their plan is to just try to sweep it under the rug until they can secure much larger government contracts. Here are a few reasons I think Microsoft will continue to just try to downplay this:

1. Microsoft is heavily reliant on government contracts and is still in the process of trying to secure the [JEDI (Joint Enterprise Defense Infrastructure) contract](https://www.defenseone.com/technology/2018/05/pentagon-wants-cloud-secure-enough-hold-nuke-secrets/148213/), which will move Department of Defense operations to a private cloud provider (for an estimated $10 billion, the largest in history).
1. Microsoft is currently bidding (and likely the front-runner) for another estimated $8 billion Department of Defense contract [DEOS (Defense Enterprise Office Solution).](https://www.nextgov.com/cio-briefing/2018/04/defense-department-pursuing-another-multibillion-dollar-cloud/147623/)
1. Trump has been very vocal talking trash about Jeff Bezos, with [Amazon being Microsoft's primary competitor](https://thehustle.co/microsoft-cloud-amazon/) for the JEDI contract, which probably makes [Microsoft think they are the preferred cloud provider](https://www.nextgov.com/it-modernization/2018/05/microsoft-announces-ic-deal-dramatically-improves-its-jedi-chances/148256/) by the current administration.
1. I don't think Microsoft has as strong of an internal culture of discussion and debate as Google does, which was crucial to enabling Google employees to [effectively organize against the Project Maven contract.](https://jacobinmag.com/2018/06/google-project-maven-military-tech-workers)
1. [this tweet](https://twitter.com/sarah_edo/status/1009286300103532545) by prominent Microsoft employee Sarah Drasner saying "I'm frustrated by the ICE situation. I'm trying to do everything I can to affect change, but I'm not sure it's going as far as I want it to.", which makes me think the internal petitions may not be as effective as we hope.

I think a walkout date will be the most effective if it is supplemented by clear demands. Maybe we could update the README to include the list of demands Microsoft employees included in [their open letter](https://int.nyt.com/data/documenthelper/46-microsoft-employee-letter-ice/323507fcbddb9d0c59ff/optimized/full.pdf):

> Acknowledging this responsibility, we request that you:
> 1. Cancel the existing Azure Government contract with ICE immediately. 
> 2. Draft, publicize, and enforce a clear policy stating that neither Microsoft nor its
contractors will work with clients who violate international human rights law.
> 3. Commit to transparency and review regarding contracts between Microsoft and
government agencies, in the US and beyond. 

I like the idea @therobinkim had about adding a section to the README to make the walk-out date opt-in. I think we can make the walkout even more effective if we show solidarity with the internal Microsoft petition by adding their demands to add to the pressure and support the employees fighting against it before there is any public resolution to the internal petition.

**Comment by rbstrachan** [comment]
@edbedbe Well put. From what you say, a walk out date is in order sooner than I thought.

I do have one question, though. If Microsoft's own employees' petition is doing little to change the situation, what more could ours do? What quality do we have that Microsoft's own employees don't?

Don't get me wrong, I'm hoping this works and I intend to honour my promise. What I'm asking is; what makes our situation so unique?

**Comment by edbedbe** [comment]
I think the Microsoft employees would probably have more leverage on the company's decision, which is why I think it would be good if we could lend our energy to support their efforts. I thought the list of demands in their open letter was very clear, and it might help them if we used the same language.

**Comment by PaulFree14** [comment]
>What quality do we have that Microsoft's own employees don't?

we're not getting payed by microsoft. it's way more easy to walk out if your not depending on a paycheck

**Comment by randolf** [comment]
@edbedbe We should definitely reference their petitions to show support.  We should continue with our current linguistic style, and resist changing it because everyone who's signed it already has signed what was written at present.  Changing the wording after they've signed it would, technically, require that they sign it again to signify agreement with what would amount to a very different petition if we were to make such a major change.

**Comment by selfagency** [comment]
FYI— I have gone forward with moving all of my repos to BitBucket. I highly encourage others to begin the process.

**Comment by randolf** [comment]
It appears that there is a link at the top of the BitBucket web site's home page to a guide to importing projects from GitHub (and GitLab) repositories:  https://www.bitbucket.org/

**Comment by makmm** [comment]
Ping! we need this. if we want to make noice, we need this.

**Comment by janriemer** [comment]
Hey guys, @selfagency 
it's now official: [export-my-github](https://github.com/janriemer/export-my-github) is here!
You can use it to export your data that you have on Github to your local hard drive.
Caution: There might be dragons :dragon:
I'd love to hear your feedback.

**Comment by jbeard4** [comment]
I wanted to mention that I have also started to archive my projects on Github and move development to Gitlab. I have already moved my main project, [SCION](https://github.com/jbeard4/SCION), and I am going to migrate my other projects soon. If anyone is interested, I wrote about this move, and what it means to me, [here](https://scion.scxml.io/blog/scion-5-release-announcement/#intention-to-migrate-from-github-to-gitlab).

**Comment by Marak** [comment]
Do we have any status on the top five signers of this petition leaving Github?

I'm starting to suspect putting these popular user names at the top of the list was more of a publicity stunt and not a serious intention to leave Github.

I'd appreciate it if these top users would make their intentions clear or be removed from the list:

 - Lea Verou @leaverou (MIT, W3C CSS Working Group, Prism, Mavo, Dabblet)
 - Sindre Sorhus @sindresorhus (AVA, XO, Awesome)
 - Thomas Fuchs @madrobby (Zepto, Script.aculo.us, Ruby on Rails)
 - Laurie Voss @seldo (LGBTQ.technology)
 - Jamie Kyle @jamiebuilds (Babel, Yarn, Flow, Parcel, Marionette, Lerna)

**Comment by def14nt** [comment]
I've dropped GitHub for Gitlab. I can confirm it;s **great**. It comes with a **WEB IDE** (basically using the barebones version of **VSCODE**). @Marak It would be great if these guys moved to Gitlab, it's better, and it would move a lot of the community over to an open source solution, which also has **FREE** private repositories.

**Comment by randolf** [comment]
@def14nt That's a great idea.  We should also keep this project open and active though so that future GitHub users can participate in it too -- late is still better than never, after all.

**Comment by def14nt** [comment]
I agree that some of these people signing up is starting to seem like a publicity stunt, but I also disagree with that for another reason.


The problem is that contributors may be unfamiliar with the GitLab/BitBucket/whatever UIs, so it isn't as easy as "swapping over". The problem is - like many other services - is the community pull GitHub has, whereas GitLab seems to have less (but that's slowly changing). This means that if one were to switch, then contributors and others might not make the jump over to the new service, rendering the project with less support from it's community.

Another problem is that links don't change themselves. If one were to delete their GitHub account, then all links to that account - and all projects from it - *would* be **dead**.

Methinks the solution here is a slow ease over to the new service, meaning make the original GitHub repo mirror the new repo, so people have chance to settle in and re-apply their stars etc.

Maybe we could start nominating people, too?

p.s I did this with [my favourite markdown editor](https://markdownlivepreview.com/), check it out :)

**Comment by selfagency** [comment]
what we actually need is a way to federate independently run git servers, regardless of what platform they run on, similarly to the way mastodon/gnu social/etc. federate content. being locked into any platform is the root of the problem which is why tim berners lee, for example, just launched solid. i would rather run my own gitea server and have a way for people with their own instances of a git server to be able to comment on issues, submit PRs, etc. without having to have an account on my server or me on theirs.

**Comment by def14nt** [comment]
You can run GitLab independently btw @selfagency

**Comment by selfagency** [comment]
i know. it would be great if gitlab, gitea, github, bitbucket, and everyone else would settle on a standard and implement it.

**Comment by selfagency** [comment]
[as of today](https://www.theverge.com/2018/10/26/17954714/microsoft-github-deal-acquisition-complete) microsoft has completed their acquisition of github. it is time to walk if you haven't already.

**Comment by def14nt** [comment]
@selfagency Is anything going to come out of all of this? The people that have signed haven't said a word. I'm starting to question if this is even a good idea.

**Comment by def14nt** [comment]
@selfagency https://blog.github.com/2018-10-26-github-and-microsoft/

**Comment by selfagency** [comment]
@def14nt there is no ethical consumption under capitalism and you gotta pick your battles. imho, i'd rather not give my business to a company that helps facilitate ethnic cleansing.

**Comment by def14nt** [comment]
@selfagency You're kind of wrong there. If big projects like Ruby on Rails, Flow, Babel, and Yarn move away, it'll look pretty bad for Github.

**Comment by def14nt** [comment]
I also use Windows 10, but give Microsoft no telemetry. Hope it puts a smile on your face :smile:

**Comment by jbeard4** [comment]
@def14nt I agree, it would have an impact if big projects were to move off of Github, however I can also see why this would be particularly challenging. In the case of big projects with multiple collaborators, they would need to reach a consensus to move development to a different platform. Transferring the code and issues to Gitlab is fairly trivial, but getting people to organize around a different infrastructure is challenging. Nobody wants to lose their Github stars, and nobody wants to create friction for developers.

Ultimately, everyone is going to have to follow their own conscience. This is true for individual contributors as well as project organizers. But at least this project asks the question of whether centralizing OSS development around a closed platform, now owned by Microsoft, with all they are doing to support ICE, is a good thing for the OSS community.

**Comment by janriemer** [comment]
@jbeard4 Totally agree!

> But at least this project asks the question of whether centralizing OSS development around a closed platform, now owned by Microsoft, with all they are doing to support ICE, is a good thing for the OSS community.

That is why we need a Github clone on the [SAFE network](https://safenetwork.tech/). The SAFE network is nothing less than a _new internet_, fully decentralized and distributed. It has even the possibility to build a **complete new economy** on the internet - gone are the days of privacy exploitation (some call it _advertising_)!
SAFE is currently in Alpha 2 with Alpha 3 on the horizon. They are also [partnering](https://safenetforum.org/t/join-the-safe-solid-crew/25819) with [Solid](https://solid.inrupt.com/).

**Comment by jbeard4** [comment]
@janriemer I'll look into this, thanks. For now, I think running a Gitlab instance on Digital Ocean, or using their hosted service at gitlab.com, is a good first step.

<img width="1280" alt="screen shot 2018-10-29 at 8 21 53 pm" src="https://user-images.githubusercontent.com/586962/47687853-b01dab80-dbb8-11e8-991b-2d54de3f2121.png">

**Comment by def14nt** [comment]
I'm waiting for Microsoft to make their first move to change Github. They're laying low right now, because it seems that acting would be bad for business, because **so** many are wary of Microsoft as a company, and a large amount of the ones that aren't have a relationship with Microsoft (a job). I can see Microsoft's involvement with Github as a very incremental one, where they tap the prices up more and more, until people retaliate. 

When that happens, we should advertise this repository and other projects related to leaving the Github platform, as that would give other small projects like GitLab and Gitea a helping hand, and it would be a great means to retaliate, as mentioned above, and it would empower the smaller open source alternatives, so that's two birds with one stone.

Additionally, can we actually arrange this leaving date? Is everyone that put their name down *actually* on board? I understand that migration is difficult, but if it is not available, then send in a pull request with your name revoked from the list.

**Comment by jbeard4** [comment]
@def14nt You might want to look at git on IPFS. https://medium.com/@alexberegszaszi/mango-git-completely-decentralised-7aef8bcbcfe6

**Comment by def14nt** [comment]
OMG! Is it really down again?!

**Comment by randolf** [comment]
@def14nt I don't think that forcing people to move is going to be helpful in garnering support.  After all, a lot of people involved in open source projects do so on a volunteer basis, and may not have time available to make the change happen (at least not immediately).

I think it's better to keep promoting this project without a definite commitment expectation (positive encouragement is certainly okay though) because that makes it easier for people, and it also helps to promote awareness in an ongoing fashion.

With regard to Microsoft making changes to GitHub, I'm concerned about this too, especially in light of their terrible track record when it comes to quality and security, and their long and not-so-friendly history of resisting closed source software (which, in my mind, makes their motives suspect).

One of the important ideas that I think this project also promotes is that Microsoft is a very profitable company that puts money before people, and it highlights ICE as a particularly dreadful real-world example of this.  This project puts a certain pressure on Microsoft as well as other organizations to not support separating innocent children from their families, and so ongoing long-term support of this resistance needs to be open and continuous because those children need to be reunited with their families immediately instead of being left in jail cells and treated like criminals that they most obviously are not.

This project is not just about moving away from GitHub to send a message to Microsoft, it's also about a much more important issue of the human rights of thousands of innocent children who have unfairly, unjustly, and forcibly been removed from their families.

**Comment by def14nt** [comment]
Might use Github and Gitlab. GitHub is better for getting support tbch

**Comment by def14nt** [comment]
Really think this should be dropped. It hasn't had a scrap of activity for a while.

@selfagency  People have lost interest, and it's time to be honest.

**Comment by selfagency** [comment]
It hasn't had a scrap of activity because a) I rarely come to GitHub anymore because I made good on my commitment to leave; b) I was banned from Twitter for standing up to white supremacists so I can no longer organize around the petition within my network; c) no one else has taken it upon themselves to reach out to and organize others. The only thing there is to be honest about is the fact that GitHub has become as ubiquitous as Google and Facebook and so much of the development world has come to depend on it, that leaving can be detrimental to one's advancement as a developer. This is a very *bad* thing and demonstrates the dangers of centralization and walled gardens which is now doubly the case for GitHub with its acquisition by Microsoft. Why you think you have any place to tell me to "drop" a principled stance against Microsoft for its participation in our racist immigration regime is beyond me but hey, go off. I also pulled everything I had off of Amazon Web Services. That I have greater conviction and am more willing to suffer the risks and consequences of leaving GitHub and others don't is on *them*, not me nor this petition. 🤷🏼‍♂️

**Comment by selfagency** [comment]
Oh, and there's going to be a feature story on this petition in the California Sunday Magazine next month. Because people have lost interest.

**Comment by def14nt** [comment]
@selfagency I didn't **tell** you to drop it. I suggested, which are two very different things. If people haven't lost interest, where's the other issues? I'm getting sick of people like you trying to twist what I'm saying, when you miss the point completely.

If you really want to keep your finger on the pulse, so to speak, on the people that have left since signing, why haven't you opened an issue with a title like `Have you quit?` (or something similar, maybe a different method), which would assert the usefulness of this petition.

**Comment by def14nt** [comment]
@selfagency Could we ask them all for an update, maybe? We need to get shit moving faster than this if we ever want to get this done.

**Comment by randolf** [comment]
@def14nt I don't think it's reasonable to expect a petition organizer to also write software to ease the transition to a particular alternative.  This petition, in my view, has not lost relevance.  Although I haven't completely stopped contributing to GitHub-hosted projects, I have reduced my activity in large part because of what this petition highlights (I don't like using certain vendor's products, but there are some instances where this isn't always a realistic option for me).

You mentioned that there are systems that depend on GitHub, and that's certainly a reasonable concern.  It's also indicative of a policy decision by some organizations to be reliant on GitHub -- what happens in the event that GitHub, for whatever reason, ceases to operate?  Do those organizations have a back-up plan that they can switch to in a timely manner?

If GitHub was fully open-source and free for download (it wasn't, the last time I looked into this), I'm confident that many groups and organizations would set up their own private instances of GitHub just for their own closed-source projects.  Some folks may argue that this would be disadvantageous for GitHub, while others may argue the opposite, and many of said groups and organizations would likely want to keep their private GitHub systems up-to-date with the new releases which would certainly speak to maintaining GitHub's relevance.

I see no problem with keeping this petition project open, and keeping it open ensures that more people who share these important concerns can sign it when they happen to encounter it.

**Comment by def14nt** [comment]
@selfagency p.s Maybe you could be a bit less abrasive to newcomers, too? I understand this is important to you, but you constantly throw around the word "trolling" when it's not always applicable.

**Comment by selfagency** [comment]
This petition has been repeatedly targeted by campaigns on 4chan, Reddit, and elsewhere for defacement and trolling. You'll forgive me if I don't give people who come here to raise objections the benefit of the doubt.

**Comment by def14nt** [comment]
#350

**Comment by def14nt** [comment]
I'm starting to see that this petition is losing attention because of these people that have signed up without any intent to leave this platform at all. I really hope this petition doesn't stall, as it was started with a good heart.

One would say it would be wise to contact some of the news sites to give this a bit of a leg up, because it looks (to me) that this petition desperately needs it.

A rolling stone gathers no moss.

**Comment by randolf** [comment]
@def0x099 The fact that people are still signing this petition is, in my opinion, a good thing.  For those who can leave this platform, I applaud.  For those who can't (e.g., their employer standardized on it and is not interested in finding an alternative), signing the petition is still an important thing to do because it shows that ongoing support persists.

I think that encouraging people to sign the petition is important, as is supporting those who decide to take their business elsewhere.  Overall, more support in any degree is always good.

**Comment by jbeard4** [comment]
Recent and related news - now Github has signed a $200k contract with ICE: https://mobile.twitter.com/evan_greer/status/1181745056698572802

**Comment by pirix-gh** [comment]
Some of the top signers receive a part of their income from GitHub, through 100ds of repos they maintain... and I hardly see them leaving, which seems quite hypocritical. If one wanted to walk out of here, **they'd just do it**.

**Comment by daemonspudguy** [comment]
I signed it, and will move personal stuff to GitLab, but I need this account for other things.

**Comment by william-silversmith** [comment]
Not sure if I'm ready to sign and leave, but one thing that might be more convincing to me is if there was a trigger upon reaching some threshold of signatures that was large enough that I would be convinced that it might have an effect rather than solely imposing a personal cost. This would also add pressure as the activation threshold nears. It also makes signing feel cheaper and easier as you only have to do something if everyone does it. The trick is setting a threshold large enough to have an effect (tens of thousands? hundreds of thousands? millions? Some estimate of active users is required.) while setting it low enough that it is achievable within X number of months or years and also finding ways to help people comply once it's reached.
